### PR TITLE
Date Time Setting and Fix Tests

### PR DESCRIPTION
**Pull request by kevinkreiser**
So `loki` tests were broken with `mjonlir` and `baldr` changes so I've fixed that so that we can eventually merge all this stuff.

Second I refactored a bit the `date_time` stuff to only be in the route requests and actually set the `date_time` on the `location`s from the request so that this doesnt need to be done in` thor`. At this point `thor` should only need to get the `date_time.type` from the request itself (which its already doing).

@gknisely look good to you?

### CODE REVIEW

**Comment by gknisely** [comment]
:shipit:

File: src/loki/route_action.cc [hunk 1]
```diff
@@ -46,11 +46,44 @@ namespace {
...
       auto costing = request.get<std::string>("costing");
       check_locations(locations.size(), max_locations.find(costing)->second);
       check_distance(reader, locations, max_distance.find(costing)->second);
 
+      //check the date stuff
+      auto date_type = request.get_optional<int>("date_time.type");
+      auto date_time_value = request.get_optional<std::string>("date_time.value");
+      if (date_type) {
+        //not yet on this
+        if(costing == "multimodal" || costing == "transit") {
```
Comment (dgearhart) [inline src/loki/route_action.cc]: This does not look correct to me - how do you know it is arrive by?

Comment (kevinkreiser) [inline src/loki/route_action.cc]: good catch, fixing

Comment (kevinkreiser) [inline src/loki/route_action.cc]: thanks again: https://github.com/valhalla/loki/commit/d09531d717b89a8c1119c34cc903f473455c8d87

**Pull request merged**